Add developer center documentation folder to doc

csubira opened the pull request:
**Update source doc branches and config file with new origin and versions/tags**:https://github.com/CartoDB/developers/issues/537

Create new folder `developer-center` inside `doc` where all the documentation (Auth API, Import API and cartoCSS) shown in the developer center is stored.

**Note:** Once merged to `master` it can be updated when developing and will be updated automatically in the dev center.

csubira commented:
This PR includes:
- New `developer-center` folder inside `docs` in order to store the public documentation for **Auth API, Import AP**I and **CartoCSS** (previously used in carto.com/docs, now deprecated and redirected to Developer Center)
- `docs/README.md` updated to describe the new folder use
- Changes done to documentation in master during this period

The outcome of this changes is already[ in production](https://carto.com/developers/),  this branch contains the same as https://github.com/CartoDB/cartodb/tree/developer-center as this was the one connected to Developer Center, the only change would me merging this folder to master so it can be updated when developing.
**Note**: After merging to master, https://github.com/CartoDB/cartodb/pull/13643 should be closed.

jgoizueta commented on doc/developer-center/import-api/guides/07-database-connectors.md:
```diff
@@ -0,0 +1,1164 @@
+
+Alternatively, you can use the Import API to connect to an external database. Your data from the external database is cached into a CARTO table by using the `connector` parameters.
+
+There are several types of database connectors that you can connect to your CARTO account.
```
:pray: I'd love to de-connect this a little:
> There are several types of database connectors that you can activate in your CARTO account.

jgoizueta commented on doc/developer-center/import-api/reference/swagger.yaml:
```diff
@@ -0,0 +1,971 @@
+      url: 'http://doc.carto.com/pet-operations.htm'
+paths:
+  '/v1/imports':
+    post:
```
I've noticed some endpoints, like this one, lacks the `parameters` section. Is this intentional? Are we aware of this? Are we lazy?

csubira replied:
I've opened a new issue to complete this information https://github.com/CartoDB/cartodb/issues/14721 so we can merge this branch to correct the typo and have the new default branch for the doc ready.

jgoizueta commented:
Hey, don't let my review disapproval prevent you from merging this if you want to, since this introduces no new defects, and the issues I mention can be handled separately.

csubira commented:
To sum up, I updated with the branch correcting the typo identified, opened an issue  #14721 for the Import API reference lack of information (because it may take longer or more people to update and review) and added a last minute commit updating older links pointing to _carto.com/docs_ pages that were not correctly redirected to the new site. So, I think it's ready to be merged.
